solr/core/src/java/org/apache/solr/cloud/CreateAliasCmd.java

```diff
@@ -45,13 +147,92 @@ public CreateAliasCmd(OverseerCollectionMessageHandler ocmh) {
-    final String canonicalCollectionsString = StrUtils.join(canonicalCollectionList, ',');
-
     ZkStateReader zkStateReader = ocmh.zkStateReader;
-    validateAllCollectionsExistAndNoDups(canonicalCollectionList, zkStateReader);
+    ZkStateReader.AliasesManager holder = zkStateReader.aliasesHolder;
+    if (!anyRoutingParams(message)) {
+      final List<String> canonicalCollectionList = parseCollectionsParameter(message.get("collections"));
+      final String canonicalCollectionsString = StrUtils.join(canonicalCollectionList, ',');
+      validateAllCollectionsExistAndNoDups(canonicalCollectionList, zkStateReader);
+      holder.applyModificationAndExportToZk(aliases -> aliases.cloneWithCollectionAlias(aliasName, canonicalCollectionsString));
+    } else {
+      final String routedField = message.getStr(ROUTING_FIELD);
+      final String routingType = message.getStr(ROUTING_TYPE);
+      final String tz = message.getStr(TZ);
+      final String start = message.getStr(START);
+      final String increment = message.getStr(ROUTING_INCREMENT);
+      final String maxFutureMs = message.getStr(ROUTING_MAX_FUTURE);
+
+      try {
+        if (0 > Long.valueOf(maxFutureMs)) {
+          throw new NumberFormatException("Negative value not allowed here");
+        }
+      } catch (NumberFormatException e) {
+        throw new SolrException(BAD_REQUEST, ROUTING_MAX_FUTURE + " must be a valid long integer representing a number " +
+            "of milliseconds greater than or equal to zero");
+      }
 
-    zkStateReader.aliasesHolder.applyModificationAndExportToZk(aliases -> aliases.cloneWithCollectionAlias(aliasName, canonicalCollectionsString));
+      // Validate we got everything we need
+      if (routedField == null || routingType == null || start == null || increment == null) {
+        SolrException solrException = new SolrException(BAD_REQUEST, "If any of " + CREATE_ROUTED_ALIAS_PARAMS +
+            " are supplied, then all of " + REQUIRED_ROUTING_PARAMS + " must be present.");
+        log.error("Could not create routed alias",solrException);
+        throw solrException;
+      }
+
+      if (!"time".equals(routingType)) {
+        SolrException solrException = new SolrException(BAD_REQUEST, "Only time based routing is supported at this time");
+        log.error("Could not create routed alias",solrException);
+        throw solrException;
+      }
+      // Check for invalid timezone
+      if(tz != null && !TimeZoneUtils.KNOWN_TIMEZONE_IDS.contains(tz)) {
+        SolrException solrException = new SolrException(BAD_REQUEST, "Invalid timezone:" + tz);
+        log.error("Could not create routed alias",solrException);
+        throw solrException;
+
+      }
+      TimeZone zone;
+      if (tz != null) {
+        zone = TimeZoneUtils.getTimeZone(tz);
+      } else {
+        zone = TimeZoneUtils.getTimeZone("UTC");
+      }
+      DateTimeFormatter fmt = DATE_TIME_FORMATTER.withZone(zone.toZoneId());
```

Comment:
No we don't want to use the TZ to format the collection name

Reply:
So if someone specifies start as an iso instant, we don't want to adjust that to match the specified zone? This would lead to start=2018-01-14T20:00:00:00.0000Z&TZ=EST&router.interval=+1DAY it would seem that the first bucket should be 2018-01-15 but this way we would be creating a first bucket for 2018-01-14 instead. That seems confusing. Solr fairly consistently takes the position that times you speak to it must be in UTC, changing that here seems awkward. We could allow more general time format and take the timezone from the entered time, so that they can enter 2018-01-15T00:00:00:00.0000EST which is unlike the rest of how solr handles time AFAIK but would make the correspondence between the collection name and the time value entered intuitive (and maybe complicates combining this with the datemath) 

I had originally proposed that collections names for this be UTC and making them full time stamps to avoid all this TZ stuff, but the response (https://issues.apache.org/jira/browse/SOLR-11299?focusedCommentId=16204466&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-16204466) was we had to make the collection names friendly for the same reason TZ is provided elsewhere, which means accounting for timezones. (this also leads to yearly double sized hourly partitions and strange mixing at sub-hour partitions, thought that resolution is probably an unusual case)

Reply:
actually amending that we do accept the format used for the collection names, but since a time stamp is valid date math we need to handle this too.

Reply:
(and that format has no timzeone in it so in theory we could treat it as local time based on TZ but at the moment I treat it as utc).

Reply:
In reading the issue referenced in the above referenced comment again, it occurs to me that what I hadn't thought about is that by having TZ parameter at the top level it does sort of imply that it applies to the whole request, and we need to handle all dates supplied either as the specified UTC or if the Z is omitted handle them as the timezone specified. I had mostly been thinking in terms of the effect of TZ on updates and users trying to optimize by querying directly to the correct collection and the calculation of "next" collections. In theory, there could be 2 TZ's one for the routing configuration router.tz and one that applies to the dates in this request (the top level TZ)... but that is probably be overkill. So now I think I understand what you've been trying to tell me... you want TZ to adjust any dates that don't include a timezone, and not adjust any that do (where the only acceptable date included TZ is utc via the Z suffix)?

Reply:
> you want TZ to adjust any dates that don't include a timezone, and not adjust any that do (where the only acceptable date included TZ is utc via the Z suffix)?

Not really because Solr doesn't accept ambiguous dates (dates without a 'Z', == UTC) so that's kind of a non-issue.  _The TimeZone is needed to correctly interpret date math (if present)_.  Oh yeah, DateRangeField can accept very dates without 'Z' but lets ignore that :-)  Nonetheless DRF is strictly UTC.

The format of the collection name is an internal detail so it almost doesn't matter.  I'd rather it be permanently UTC interpreted to avoid problems in changing TZ somehow.  But I can sympathize that there's an argument to be made it would look nicer if it honored the TZ.  But again it's internal.

Reply:
An offline chat reveals that I misunderstood the intent some time back. Names for collections ARE meant to be in UTC, that will make things simpler I think.